lillisp.go

```diff
@@ -3,15 +3,78 @@ package main
 import (
 	"bufio"
 	"fmt"
+	"log"
 	"os"
+	"strconv"
 	"unicode/utf8"
 )
 
+// Atom ...
+type Atom interface{}
+
+// Pair represents the fundamental "cons-pair" in LISP-like languages
 type Pair struct {
 	car interface{}
 	cdr interface{}
 }
 
+var funcMap = map[string]func(Atom, Atom) Atom{
+	"+": func(a Atom, b Atom) Atom { return toInt(a) + toInt(b) },
```

Comment:
can we convert numbers earlier like in the scanner or process item?

Reply:
MMmaaaayyyybe...